pravega-k8-log-collection/pravega-k8s-log.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+mkdir -p /tmp/logdir
+output_dir="/tmp/logdir"
+kubectl get po,svc,pvc --all-namespaces > $output_dir/kubectl-get-po-svc-pvc-all-namespace_output
+IFS=$'\n'
+for i in $(cat $output_dir/kubectl-get-po-svc-pvc-all-namespace_output | grep -i pravega)
+do
+echo $i
+if [[ $i == *"po"* ]];
+then 
+    if [ `echo $i | awk '{print $5}'` != 0 ]
+        then     
+            kubectl -n `echo $i | awk '{print $1}'` logs `echo $i | awk '{print $2}'` -p > $output_dir/log_`echo $i | awk 'gsub(/\//,"_") {print $2}'`_previous.log
+    fi
+    kubectl -n `echo $i | awk '{print $1}'` logs `echo $i | awk '{print $2}'` > $output_dir/log_`echo $i | awk 'gsub(/\//,"_") {print $2}'`_recent.log
+    kubectl -n `echo $i | awk '{print $1}'` exec -it `echo $i |  awk '-F'[/] '{print $2}'|awk '{print $1}'` -- ls /opt/pravega/logs > /dev/null 2>&1
+    if [ $? == 0 ]; 
```

Comment:
Include the above command into this `if` statement to prevent the script from failing if you set the shell flags:

```
if [ kubectl -n `echo $i | awk '{print $1}'` exec -it `echo $i |  awk '-F'[/] '{print $2}'|awk '{print $1}'` -- ls /opt/pravega/logs > /dev/null 2>&1 ];
```

Reply:
+1

Reply:
Thanks for the suggestion, we used following as discussed:
` has_logs=$(kubectl -n `echo $i | awk '{print $1}'` exec -it `echo $i |  awk '-F'[/] '{print $2}'|awk '{print $1}'` -- ls -l /opt/pravega/logs 2>/dev/null | wc -l || echo "")
if [ $has_logs -ge 2 ]; then`